File: src/main/java/za/ac/cput/service/publisherService/PublisherService.java
package za.ac.cput.service.publisherService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import za.ac.cput.domain.Author;
import za.ac.cput.domain.Publisher;
import za.ac.cput.domain.WishList;
import za.ac.cput.factory.PublisherFactory;
import za.ac.cput.repository.AuthorRepository;
import za.ac.cput.repository.PublisherRepository;

import java.util.List;
import java.util.Optional;

@Service
public class PublisherService implements IPublisherService {

    private PublisherRepository repository;


    @Autowired
    public PublisherService(PublisherRepository repository) {
        this.repository = repository;
    }

    @Override
    public Publisher create(Publisher p) {
        Publisher publisher = PublisherFactory.buildPublisher(
                p.getName(),
                p.getYearFounded()
        );
        return repository.save(publisher);
    }

    @Override
    public Publisher read(Long aLong) {
        return repository.findById(aLong).orElse(null);
    }


    @Override
    public Publisher update(Publisher publisher) {
        return repository.save(publisher);
    }

    @Override
    public boolean delete(Long aLong) {
        repository.deleteById(aLong);
        return !repository.existsById(aLong);
    }

    @Override
    public List<Publisher> getall() {
        return repository.findAll();
    }

    public List<Publisher> findByName(String name) {
        return repository.findByNameContainingIgnoreCase(name);
    }
}

File: src/frontend/src/services/PublisherService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/comiccity/Publisher';

export const createPublisher = (publisher) => {
    return axios.post(`${API_URL}/create`, publisher);
};

export const getPublisher = (id) => {
    return axios.get(`${API_URL}/read/${id}`);
};

export const updatePublisher = (publisher) => {
    return axios.post(`${API_URL}/update`, publisher);
};

export const deletePublisher = (publisherId) => {
    return axios.delete(`${API_URL}/delete/${publisherId}`);
};

export const getAllPublishers = () => {
    return axios.get(`${API_URL}/getall`);
};
export const findPublishersByName = (name) => {
    return axios.get(`${API_URL}/findByName/${name}`);
};

